build/android/app/src/main/java/net/minetest/minetest/UnzipService.java

```diff
@@ -56,7 +56,7 @@ public UnzipService() {
 	}
 
 	private void isDir(String dir, String location) {
-		File f = new File(location + dir);
+		File f = new File(location + File.separator + dir);
```

Comment:
you can use [multiple arguments](https://docs.oracle.com/javase/7/docs/api/java/io/File.html) to build path names

```suggestion
		File f = new File(location, dir);
```

Reply:
Thanks. Forgot about it.